### PR TITLE
explain_weights in Pipelines: minimal version

### PR DESCRIPTION
**Pull request by jnothman**
It looks like eli5 had a recent release, so it's a pity to come in late for it. But here's a more minimal version of #158, to which more can be added as it is designed and tested.

### CODE REVIEW

File: eli5/sklearn/transform.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+"""transform_feature_names implementations for scikit-learn transformers
+"""
+
+import numpy as np  # type: ignore
+from sklearn.pipeline import Pipeline
```
Comment (kmike) [inline eli5/sklearn/transform.py]: it seems ` # type: ignore` should fix mypy build

**Comment by kmike** [comment]
Looks good, thanks for the persistence @jnothman!

File: docs/source/autodocs/eli5.rst [hunk 2]
```diff
@@ -13,3 +13,5 @@ The following functions are exposed to a top level, e.g.
 .. autofunction:: eli5.show_weights
 
 .. autofunction:: eli5.show_prediction
+
+.. autofunction:: eli5.transform_feature_names
```
Comment (kmike) [inline docs/source/autodocs/eli5.rst]: Yeah, I was also unsure; do you think this function will be used stand-alone, not as an implementation detail of how to make explain.. / show.. functions work?

**Comment by jnothman** [comment]
It's not an implementation detail insofar as users will need to register
their own implementations as they write their own transformers.

That's not to say that it's likely to be called outside of explain_weights.

**Comment by jnothman** [comment]
In any case, documentation, of all things, can change later.

File: eli5/sklearn/explain_weights.py [hunk 3]
```diff
@@ -422,3 +424,17 @@ def _features(target_id):
             method='linear model',
             is_regression=True,
         )
+
+
+@explain_weights.register(Pipeline)
```
Comment (kmike) [inline eli5/sklearn/explain_weights.py]: After https://github.com/TeamHG-Memex/eli5/pull/170 it should be just `@register`, so that it is registered both for global and sklearn-specific explain functions.

**Comment by kmike** [comment]
Thanks @jnothman! I'm going to make a new release soon - do you want to add anything else to it?

**Comment by jnothman** [comment]
we'll see how time goes.

**Comment by jnothman** [comment]
thanks for the review and merge!
